maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -874,9 +873,17 @@ private static <T> List<T> reverse(List<T> list) {
     }
 
     private List<File> parseExtClasspath(CliRequest cliRequest) {
-        String extClassPath = cliRequest.userProperties.getProperty(EXT_CLASS_PATH);
+        String extClassPath = cliRequest.userProperties.getProperty(Constants.MAVEN_EXT_CLASS_PATH);
         if (extClassPath == null) {
-            extClassPath = cliRequest.systemProperties.getProperty(EXT_CLASS_PATH);
+            extClassPath = cliRequest.systemProperties.getProperty(Constants.MAVEN_EXT_CLASS_PATH);
+            if (extClassPath != null) {
+                slf4jLogger.warn(
+                        "The property {} has been set using a JVM system property which is deprecated. "
+                                + "The property can be passed as a maven argument or in the maven project configuration file,"
+                                + "usually located at {}.",
+                        Constants.MAVEN_EXT_CLASS_PATH,
+                        "[rootDirectory]/.mvn/maven.properties");
```

Comment:
This message should use prop syntax: `"${rootDirectory}/.mvn/maven.properties"}`, no?

Reply:
Makes sense.